lib/explorer/series.ex

```diff
@@ -108,15 +108,19 @@ defmodule Explorer.Series do
         [1.0, 2.0]
       >
 
+  Trying to create a "nil" series will result in a series of floats.
+  If this is not desired, you can always use `cast/2` to change the type.
+
+      iex> Explorer.Series.from_list([nil, nil])
```

Comment:
Can't there be an option that already accepts the target type?

Reply:
Yeah, I think this is a good idea :+1: 
The `PolarsBackend.Series` accept a `dtype` option. We could accept in the main function as well.

Reply:
Yep totally agree @polvalente!

Reply:
I found this to be harder than I thought :grimacing: 
We would first have to cast the values from series to the desired type before passing to the backend. Or we would have to cast in the backend. (this is the case for setting the `dtype` of series with nils and not nil).

What I think it's easier to do now is to control only the case where the list is "all nil" and we want to specify a dtype for the column. WDYT?

Reply:
Perhaps if the target type is passed, and it's different than the inferred one, we could use `cast/2` internally, for ease of use

Reply:
Oh, good idea! I changed to do something like that. Please check again.